model/project_ref.go

```diff
@@ -1255,6 +1315,176 @@ func (projectRef *ProjectRef) Upsert() error {
+func saveProjectRefForSection(projectId string, p *ProjectRef, section ProjectRefSection) (bool, error) {
+	if p == nil {
+		p = &ProjectRef{} // use a blank project ref to default the section to repo
+	}
+	var err error
+	switch section {
+	case ProjectRefGeneralSection:
+		err = db.Update(ProjectRefCollection,
+			bson.M{ProjectRefIdKey: projectId},
+			bson.M{
+				"$set": bson.M{
+					ProjectRefEnabledKey:                 p.Enabled,
+					ProjectRefBatchTimeKey:               p.BatchTime,
+					ProjectRefRemotePathKey:              p.RemotePath,
+					projectRefSpawnHostScriptPathKey:     p.SpawnHostScriptPath,
+					projectRefDispatchingDisabledKey:     p.DispatchingDisabled,
+					ProjectRefDeactivatePreviousKey:      p.DeactivatePrevious,
+					projectRefRepotrackerDisabledKey:     p.RepotrackerDisabled,
+					projectRefDefaultLoggerKey:           p.DefaultLogger,
+					projectRefCedarTestResultsEnabledKey: p.CedarTestResultsEnabled,
+					projectRefPatchingDisabledKey:        p.PatchingDisabled,
+					projectRefTaskSyncKey:                p.TaskSync,
+					ProjectRefDisabledStatsCacheKey:      p.DisabledStatsCache,
+					ProjectRefFilesIgnoredFromCacheKey:   p.FilesIgnoredFromCache,
+				},
+			})
+	case ProjectRefAccessSection:
+		err = db.Update(ProjectRefCollection,
+			bson.M{ProjectRefIdKey: projectId},
+			bson.M{
+				"$set": bson.M{
+					ProjectRefPrivateKey:    p.Private,
+					ProjectRefRestrictedKey: p.Restricted,
+					ProjectRefAdminsKey:     p.Admins,
+				},
+			})
+	case ProjectRefGithubAndCQSection:
+		err = db.Update(ProjectRefCollection,
+			bson.M{ProjectRefIdKey: projectId},
+			bson.M{
+				"$set": bson.M{
+					projectRefPRTestingEnabledKey:      p.PRTestingEnabled,
+					projectRefGithubChecksEnabledKey:   p.GithubChecksEnabled,
+					projectRefGithubTriggerAliasesKey:  p.PatchTriggerAliases,
+					projectRefGitTagVersionsEnabledKey: p.GitTagVersionsEnabled,
+					ProjectRefGitTagAuthorizedUsersKey: p.GitTagAuthorizedUsers,
+					ProjectRefGitTagAuthorizedTeamsKey: p.GitTagAuthorizedTeams,
+					projectRefCommitQueueKey:           p.CommitQueue,
+				},
+			})
+	case ProjectRefNotificationsSection:
+		err = db.Update(ProjectRefCollection,
+			bson.M{ProjectRefIdKey: projectId},
+			bson.M{
+				"$set": bson.M{projectRefNotifyOnFailureKey: p.NotifyOnBuildFailure},
+			})
+	case ProjectRefWorkstationsSection:
+		err = db.Update(ProjectRefCollection,
+			bson.M{ProjectRefIdKey: projectId},
+			bson.M{
+				"$set": bson.M{projectRefWorkstationConfigKey: p.WorkstationConfig},
+			})
+	case ProjectRefTriggersSection:
+		err = db.Update(ProjectRefCollection,
+			bson.M{ProjectRefIdKey: projectId},
+			bson.M{
+				"$set": bson.M{
+					projectRefTriggersKey: p.Triggers,
+				},
+			})
+	case ProjectRefAliasSection:
+		err = db.Update(ProjectRefCollection,
+			bson.M{ProjectRefIdKey: projectId},
+			bson.M{
+				"$set": bson.M{
+					projectRefPatchTriggerAliasesKey: p.PatchTriggerAliases,
```

Comment:
I think that we decided to put patchTriggerAliases in the github section.

Reply:
Sorry I should've taken notes on our conversation. Didn't we decide to put it in the same section it relied on? (Which I thought was this one because we defined the alias here and then used it in patch trigger aliases but I'm pretty confused now). I'm going to ping Lara and see what she wrote down.

Reply:
I talked to Lara, she also remembers it being in this section for the same reason but can't find her notes on the subject. We may need to re have the design conversation with her to iron it back out; for now let's leave it here and we can re-visit if necessary.